File: src/immersed.dive.shop.model/EventDate.cs
using System;

namespace immersed.dive.shop.model
{
    public class EventDate : IEntity
    {
        public EventDate()
        {
            
        }

        public DateTime Date { get; set; }
        public int EstimatedDuration { get; set; }
        public Guid Id { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime LastUpdated { get; set; }
        public bool Live { get; set; }
    }
}

File: the-client-app/src/app/models/eventDate.ts
export interface EventDate {
    id?: string;    
    date?: Date;
    estimatedDuration?: number;
    dateCreated?: Date;
    lastUpdated?: Date;
    live?:boolean;    
}

export class EventDate implements  EventDate {
    constructor(eventDate?: EventDate){
        if(eventDate) {
            this.id = eventDate.id;
            this.date = eventDate.date;
            this.estimatedDuration = eventDate.estimatedDuration;            
            this.dateCreated = eventDate.dateCreated;
            this.lastUpdated = eventDate.lastUpdated;
            this.live =eventDate.live;
        }
    }
}
